lib/runtime/wasmer/imports.go

```diff
@@ -567,51 +500,57 @@ func ext_crypto_sr25519_generate_version_1(context unsafe.Pointer, keyTypeID C.i
 		panic(err)
 	}
 
-	ks, err := runtimeCtx.Keystore.GetKeystore(id)
+	ks, err := instanceContext.Keystore.GetKeystore(id)
 	if err != nil {
 		logger.Warnf("error for id "+common.BytesToHex(id)+": %s", err)
-		return 0
+		return []wasmer.Value{wasmer.NewI32(0)}, nil
 	}
 
 	err = ks.Insert(kp)
 	if err != nil {
 		logger.Warnf("failed to insert key: %s", err)
-		return 0
+		return []wasmer.Value{wasmer.NewI32(0)}, nil
 	}
 
 	ret, err := toWasmMemorySized(instanceContext, kp.Public().Encode())
 	if err != nil {
 		logger.Errorf("failed to allocate memory: %s", err)
-		return 0
+		return []wasmer.Value{wasmer.NewI32(0)}, nil
 	}
 
+	// TODO dont think this is safe
```

Comment:
Looking for feedback on this type cast

Reply:
It seems that currently wasmer only allows type int to be passed in for I32 wasm type. I have opened an issue asking about this [here](https://github.com/wasmerio/wasmer-go/issues/382).

As discussed offline, I am going to add checks in all instances where we cast values to send to the runtime.